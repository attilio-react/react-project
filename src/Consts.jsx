const consts = { 
    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1
}

export {consts}
