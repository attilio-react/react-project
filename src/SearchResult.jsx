import React from 'react';

import {SearchContext} from './SearchContext.jsx'

import {NoResult} from './NoResult.jsx'
import {ResultList} from './ResultList.jsx'

class SearchResult extends React.PureComponent {
  render () {
    return (<SearchContext.Consumer>
          {results => {
            const resultsFound = results && results.items && results.items.length > 0;
            return resultsFound ? <ResultList results={results.items}/> : <NoResult/>;
          }}
    </SearchContext.Consumer>)
  }
}


export {SearchResult}


