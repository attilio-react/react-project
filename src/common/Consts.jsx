const consts = { 
    SEARCH_SCREEN : 0,
    DETAIL_SCREEN: 1,


    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1,


    PARAM_SORT_BY: 'sortBy',
    PARAM_SORT_ORDER: 'sortOrder',
    PARAM_SEARCH: 'search',
    PARAM_SEARCH_BY: 'searchBy',

    VALUE_BY_GENRE: 'genres',
    VALUE_BY_TITLE: 'title',
    VALUE_BY_RELEASE_DATE: 'release_date',
    VALUE_BY_VOTE_AVERAGE: 'vote_average',

    VALUE_ASC: 'asc',
    VALUE_DESC: 'desc'
}

export {consts}
