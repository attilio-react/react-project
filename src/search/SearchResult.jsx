import React from 'react';

import {SearchContext} from './SearchContext.jsx'

import {NoResult} from './NoResult.jsx'
import {ResultContainer} from './ResultContainer.jsx'

class SearchResult extends React.PureComponent {
  render () {
    return (<SearchContext.Consumer>
          {results => {
            const resultsFound = results && results.items && results.items.length > 0;
            return resultsFound ? <ResultContainer /> : <NoResult/>;
          }}
    </SearchContext.Consumer>)
  }
}


export {SearchResult}


