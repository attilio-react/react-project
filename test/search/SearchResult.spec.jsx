import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {SearchResult} from '|search/SearchResult.jsx';
import {SearchContext} from '|search/SearchContext.jsx';
import {consts} from '|common/Consts.jsx'

const CONTEXT = {
    itemClickCb: jest.fn(),
    items: [
	{
	    id: 1,
	    poster_path: 'http://hello.com/poster.png',
	    title: 'Movie Title',
	    vote_average: 4.4,
	    release_date: '2011-01-01',
	    runtime: 129,
	    overview: 'This is a movie about ....',
	    genres: ['Horror', 'Thriller']
	},
	{
	    id: 2,
	    poster_path: 'http://hello.com/poster2.png',
	    title: 'Another Movie Title',
	    vote_average: 4.4,
	    release_date: '2012-02-02',
	    runtime: 139,
	    overview: 'This is another movie about ....',
	    genres: ['War', 'History']
	}
    ],
    total: 2
}

describe('SearchResult rendering', () => {
  it('Should render with elements', () => {
    const 
        component = renderer.create(
            <SearchContext.Provider value={CONTEXT}>
               <SearchResult />
            </SearchContext.Provider>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Should render without elements', () => {
    const 
        component = renderer.create(
            <SearchContext.Provider value={{...CONTEXT, items: []}}>
               <SearchResult />
            </SearchContext.Provider>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});


