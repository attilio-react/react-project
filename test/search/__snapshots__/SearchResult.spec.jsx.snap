// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult rendering Should render with elements 1`] = `
<div>
  <div>
    <span>
      2 movies found
    </span>
           
    <span>
      Sort by: 
    </span>
     
    <input
      style={
        Object {
          "textDecoration": "none",
        }
      }
      type="button"
      value="release date"
    />
     
    <input
      style={
        Object {
          "textDecoration": "underline",
        }
      }
      type="button"
      value="rating"
    />
  </div>
  <div>
    <div
      style={
        Object {
          "border": "1px solid black",
        }
      }
    >
      <a
        onClick={[Function]}
      >
        <p>
          <img
            src="http://hello.com/poster.png"
          />
        </p>
        <p>
          <span>
            2011-01-01
          </span>
           
          <span>
            Horror, Thriller
          </span>
        </p>
        <p>
          <span>
            Movie Title
          </span>
        </p>
      </a>
    </div>
    <div
      style={
        Object {
          "border": "1px solid black",
        }
      }
    >
      <a
        onClick={[Function]}
      >
        <p>
          <img
            src="http://hello.com/poster2.png"
          />
        </p>
        <p>
          <span>
            2012-02-02
          </span>
           
          <span>
            War, History
          </span>
        </p>
        <p>
          <span>
            Another Movie Title
          </span>
        </p>
      </a>
    </div>
  </div>
</div>
`;

exports[`SearchResult rendering Should render without elements 1`] = `
<div
  style={
    Object {
      "fontSize": "large",
    }
  }
>
  No films found
</div>
`;
